proto/babylon/btcstkconsumer/v1/tx.proto

```diff
@@ -52,6 +52,9 @@ message MsgRegisterConsumer {
   string consumer_name = 3;
   // consumer_description is a description for the consumer (can be empty)
   string consumer_description = 4;
+  // eth_l2_finality_contract_address is the address of the finality contract
+  // (if set, then this means this is an ETH L2 integration)
+  string eth_l2_finality_contract_address = 5 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```

Comment:
This is a little obscure. This is the Cosmos address of the "assistant" finality contract in the Babylon chain for that Eth-L2 consumer, right? Better to clarify all that in the comments.

Again, what about making `MsgRegisterConsumer` a `oneof` here? Not sure this is a good idea, or even possible, just saying.